releases/development/master/extra/install-chef-suse.sh

```diff
@@ -500,47 +524,70 @@ check_repo_content () {
     fi
 }
 
-check_repo_product () {
-    repo="$1" expected_summary="$2" create_if_missing="$3"
-    products_xml=/srv/tftpboot/repos/$repo/repodata/products.xml
-    if ! grep -q "<summary>$2</summary>" $products_xml; then
-        if skip_check_for_repo "$repo"; then
-            echo "Ignoring failed repo check for $repo due to \$REPOS_SKIP_CHECKS ($products_xml is missing summary '$expected_summary')"
-            if [ ! -d /srv/tftpboot/repos/$repo -a "x$create_if_missing" != "xfalse" ]; then
-                echo "Creating repo skeleton to make AutoYaST happy."
-                mkdir /srv/tftpboot/repos/$repo
-                /usr/bin/createrepo /srv/tftpboot/repos/$repo
-            fi
-            return 0
-        fi
-        die "$repo does not contain the right repository ($products_xml is missing summary '$expected_summary')"
+check_repo_key () {
+    version="$1" repo="$2"
+    repomd_key=/srv/tftpboot/suse-$version/repos/$repo/repodata/repomd.xml.key
+    if [ ! -f $repomd_key ]; then
+      die "$repo ($version) does not seem to be signed"
+    else
+      case "`md5sum $repomd_key | awk '{print $1}'`" in
+        a0857768a900ae6bb1e6c8af62460ce3|b2740847428d13d201642ca9cf89bec7)
+          # SLE11 keys
+          ;;
+        9a62177e1c6852d48453ed3909956d6b)
+          # SLE12 key
+          ;;
+        6e2920076653964b7de9d6d0421955bb)
+          # Devel:Cloud key
+          ;;
+        *)
+          die "$repo ($version) does not seem to be signed with the right key"
+          ;;
+      esac
     fi
 }
 
-# FIXME: repos that we cannot check yet:
-#   SP3-Updates is lacking products.xml
-#   Cloud: we don't have the final md5
-REPOS_SKIP_CHECKS+=" Cloud SLES11-SP3-Updates SUSE-Cloud-5-Pool SUSE-Cloud-5-Updates"
+check_repo_tag () {
+    tag="$1" version="$2" repo="$3" expected="$4" create_if_missing="$5"
 
-# HAE add-on should remain optional for now
-REPOS_SKIP_CHECKS+=" SLE11-HAE-SP3-Pool SLE11-HAE-SP3-Updates"
+    repo_dir=/srv/tftpboot/suse-$version/repos/$repo
 
-# Storage should remain optional for now
-REPOS_SKIP_CHECKS+=" SUSE-Enterprise-Storage-1.0-Pool SUSE-Enterprise-Storage-1.0-Updates"
+    case "$tag" in
+      summary)
+        xml=$repo_dir/repodata/products.xml
+        ;;
+      repo)
+        xml=$repo_dir/repodata/repomd.xml
+        ;;
+      *)
+        die "Internal error: unknown tag $tag for check_repo_tag"
+        ;;
+     esac
 
-MEDIA=/srv/tftpboot/suse-11.3/install
+    ignore_failure=1
+    skip_check_for_repo "$repo" || ignore_failure=0
```

Comment:
Isn't this reversed? It seems that if $repo **is** marked for check skip, than you may hit that ``die`` later